### PR TITLE
Style Guide Auto-Suggest and PR template

### PR DESCRIPTION
**Pull request by palday**
- This updates our format enforcer to auto-suggest corrections as PR review comments. I've left `tests` out for now because we never enforced style guides there previously and there are a lot of changes if included. I have left `docs` in because there are only two small changes there and it shows off the new action.
- I've added a Blue Style badge to the README
- ~I've removed the Tier 2 badge from the README because we currently don't have any Tier 2 platforms declared. We should probably revisit this at some point and maybe change the badges to "LTS, current release, nightly".~ I've changed the CI runs, badges, and supported platforms to indicate that we test all 3 big platforms for LTS, but only Ubuntu for current release and nightly.
- I've added a step to CI that cancels previous runs that are still in progress.
- I've  added a PR template reminding us to update the NEWS entry. :smile:

### CODE REVIEW

**Comment by dmbates** [comment]
So, why do you think contributors would fail to update the `NEWS.md` file? :smiley:

File: docs/make.jl [hunk 1]
```diff
@@ -3,7 +3,6 @@ using StatsBase
 using MixedModels
 
 makedocs(
-#    root = joinpath(dirname(pathof(MixedModels)), "..", "docs"),
     sitename = "MixedModels",
     doctest = true,
     pages = [
```
[the remark this answers is not in the document]
Comment (dmbates) [reply docs/make.jl:8]: Good idea.

**Review by dmbates — APPROVED** [review]
Looks good, although I am not sure what the failure in the check means.  I trust you will wrestle with that one.  Is it just missing the semicolon?

**Comment by palday** [comment]
@dmbates the format check fails whenever it makes suggestions (in this case: the semicolon and spacing around `=`). So when we accept the suggestions, it re-runs and everything will pass. Doing that now :smile:

**Comment by dmbates** [comment]
@palday Great!  Thanks for doing this.

**Pull request merged**